File: 9.回文数.c
/*
 * @lc app=leetcode.cn id=9 lang=c
 *
 * [9] 回文数
 */

// @lc code=start
bool isPalindrome(int x) {
    if(x<0){
        return false;
    }
    long y=x,z=0;
    while(y){
        z=z*10+y%10;
        y/=10;
    }
    return x==z;
}
// @lc code=end

File: 9.回文数.py
#
# @lc app=leetcode.cn id=9 lang=python3
#
# [9] 回文数
#

# @lc code=start
class Solution:
    def isPalindrome(self, x: int) -> bool:
        # x=str(x)
        # x_reverse=x[::-1]
        # return x==x_reverse
        return x>=0 and str(x)==str(x)[::-1]
# @lc code=end

